sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonParser.scala

```diff
@@ -419,7 +419,10 @@ class JacksonParser(
     val row = new GenericInternalRow(schema.length)
     var badRecordException: Option[Throwable] = None
     var skipRow = false
-
+    // Apply default values from the column metadata to the initial row, if any.
+    for ((value: Any, i: Int) <- schema.existenceDefaultValues.zipWithIndex) {
```

Comment:
No biggie but maybe we could add an if-else to skip this assignment if there are no default values (since arguably most cases won't have default values, and this is performance sensitive code path)

Reply:
SG, done.